plugins/scanners/fossid/src/main/kotlin/FossIdNamingProvider.kt

```diff
@@ -58,15 +61,42 @@ class FossIdNamingProvider(
 
     fun createScanCode(projectName: String, deltaTag: FossId.DeltaTag? = null, branch: String = ""): String {
         var defaultPattern = "#projectName_#currentTimestamp"
-        val builtins = mutableMapOf("#projectName" to projectName, "#branch" to branch)
+        val builtins = mutableMapOf("#projectName" to projectName)
 
         deltaTag?.let {
             defaultPattern += "_#deltaTag"
             builtins += "#deltaTag" to deltaTag.name.lowercase()
         }
 
-        val pattern = namingScanPattern ?: defaultPattern
-        return replaceNamingConventionVariables(pattern, builtins, namingConventionVariables)
+        if (branch.isNotBlank()) {
+            val branchName = normalizeBranchName(branch, defaultPattern, builtins)
+            defaultPattern += "_#branch"
+            builtins += "#branch" to branchName
+        }
+
+        return replaceNamingConventionVariables(
+            namingScanPattern ?: defaultPattern, builtins, namingConventionVariables
+        )
+    }
+
+    /**
+     * Replaces non-standard characters in branch name and trimming it's length to one that will not exceed
+     * maximum length of FossID scan ID, when combined with rest of variables
+     */
+    private fun normalizeBranchName(
+        branch: String,
+        defaultPattern: String,
+        scanCodeVariables: Map<String, String>
+    ): String {
+        val noBranchScanCodeLength =
+            replaceNamingConventionVariables(
+                namingScanPattern ?: defaultPattern,
+                scanCodeVariables,
+                namingConventionVariables
+            ).length
+
+        val maxBranchNameLength = MAX_SCAN_CODE_LEN - noBranchScanCodeLength
```

Comment:
With this code, what happens if someone put a pattern in `namingScanPattern`  which leads to a `noBranchScanCodeLength` > 254 ?

Reply:
Good point. Fixed.